File: calculator.py
#!/usr/bin/env python3
import math
import sys

MENU = '''
Scientific Calculator
---------------------
1) Square root (√x)
2) Factorial (!x)
3) Natural log (ln x)
4) Power (x^b)
5) Exit
Choose an option (1-5): '''

def sqrt(x):
    if x < 0:
        raise ValueError("sqrt: negative input")
    return math.sqrt(x)

def factorial(n):
    if n < 0:
        raise ValueError("factorial: negative input")
    if not float(n).is_integer():
        raise ValueError("factorial: non-integer input")
    return math.factorial(int(n))

def ln(x):
    if x <= 0:
        raise ValueError("ln: non-positive input")
    return math.log(x)

def power(x, b):
    return math.pow(x, b)

def main():
    while True:
        try:
            choice = input(MENU).strip()
            if choice == '1':
                v = float(input('Enter x: '))
                print('√{} = {}'.format(v, sqrt(v)))
            elif choice == '2':
                v = float(input('Enter n (integer): '))
                print('{}! = {}'.format(int(v), factorial(v)))
            elif choice == '3':
                v = float(input('Enter x: '))
                print('ln({}) = {}'.format(v, ln(v)))
            elif choice == '4':
                x = float(input('Enter base x: '))
                b = float(input('Enter exponent b: '))
                print('{}^{} = {}'.format(x, b, power(x, b)))
            elif choice == '5':
                print('Goodbye')
                sys.exit(0)
            else:
                print('Invalid option')
        except Exception as e:
            print('Error:', e)

if __name__ == '__main__':
    main()

File: src/main/java/com/example/Calculator.java
package com.example;
import java.util.Scanner;

public class Calculator {

    public static double squareRoot(double x) {
        return Math.sqrt(x);
    }

    public static long factorial(int n) {
        if (n < 0) throw new IllegalArgumentException("Negative number");
        long result = 1;
        for (int i = 1; i <= n; i++) result *= i;
        return result;
    }

    public static double naturalLog(double x) {
        if (x <= 0) throw new IllegalArgumentException("Log undefined for non-positive numbers");
        return Math.log(x);
    }

    public static double power(double x, double b) {
        return Math.pow(x, b);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (true) {
            System.out.println("\n==== Scientific Calculator ====");
            System.out.println("1. Square Root");
            System.out.println("2. Factorial");
            System.out.println("3. Natural Logarithm");
            System.out.println("4. Power");
            System.out.println("5. Exit");
            System.out.print("Enter choice: ");
            int choice = sc.nextInt();

            switch (choice) {
                case 1:
                    System.out.print("Enter number: ");
                    double n1 = sc.nextDouble();
                    System.out.println("√" + n1 + " = " + squareRoot(n1));
                    break;
                case 2:
                    System.out.print("Enter integer: ");
                    int n2 = sc.nextInt();
                    System.out.println(n2 + "! = " + factorial(n2));
                    break;
                case 3:
                    System.out.print("Enter number: ");
                    double n3 = sc.nextDouble();
                    System.out.println("ln(" + n3 + ") = " + naturalLog(n3));
                    break;
                case 4:
                    System.out.print("Enter base: ");
                    double base = sc.nextDouble();
                    System.out.print("Enter exponent: ");
                    double exp = sc.nextDouble();
                    System.out.println(base + "^" + exp + " = " + power(base, exp));
                    break;
                case 5:
                    System.out.println("Exiting...");
                    sc.close();
                    System.exit(0);
                default:
                    System.out.println("Invalid choice");
            }
        }
    }
}
